numpy/typing.pyi

```diff
@@ -23,7 +23,7 @@ class _DtypeDictBase(TypedDict):
 # Mandatory + optional keys
 class _DtypeDict(_DtypeDictBase, total=False):
     offsets: Sequence[int]
-    titles: Sequence[Union[bytes, Text, None]]
+    titles: Sequence[Any]  # Elements should be valid as dict keys
```

Comment:
This comment isn't accurate. _If_ the elements are `str`, then they end up in `dtype.fields`. If they're not, then they don't.

There are no requirements on hashability, nor will numpy even attempt to hash them if they are not `str`.

Perhaps
```suggestion
    titles: Sequence[Any]  # Only `str` elements are usable as indexing aliases, but all objects are legal
```

Reply:
Ah, you're right. After reading the [structured arrays](https://numpy.org/doc/stable/user/basics.rec.html#field-titles) documentation I was under the impression that passed values would always serve as dict `dtype.fields` keys.

Still, this does beg the question whether or not there is actually any legit use for non-`str` keys here. Do they actually serve a purpose or are they simply ignore?
If it's the latter case, shouldn't we actively discourage their use?

Reply:
Any thoughts on the concern I raised above?

Reply:
Updated the respective comment (https://github.com/numpy/numpy/pull/16622/commits/b4fa0b966732124afce2d19b460ee35a5b2ce124).